src/main/kotlin/StringHelper.kt

```diff
@@ -34,4 +35,33 @@ object StringHelper {
     enum class MessageTypes {
         MISSING_PERMISSIONS
     }
+
+    private val rulesMap = HashMap<Long, HashMap<String, String>>()
+
+    fun removeServerRules(server: Server) {
+        rulesMap.remove(server.id)
+    }
+
+    suspend fun getRule(server: Server, rule: String): String? {
+        if (rulesMap[server.id] == null) rulesMap[server.id] = HashMap()
+        if (rulesMap[server.id]!!.isEmpty()) {
+            val noFormatRules =
+                (server.textChannels.findByName("rules") ?: return null).getMessages().last().content
+            val rulesByLine = noFormatRules.split("\n")
+            var lastNumber = ""
+            for (ruleLine in rulesByLine) {
+                val ruleSplit = ruleLine.trim().split(" ", limit = 2)
+                var ruleID = ruleSplit[0].replace("\\*|\\.".toRegex(), "")
+                if (ruleID.isEmpty()) continue
+                try {
+                    Integer.parseInt(ruleID)
+                    lastNumber = ruleID
+                } catch (_: NumberFormatException) {
+                    ruleID = "$lastNumber$ruleID"
+                }
+                rulesMap[server.id]!![ruleID] = ruleSplit[1]
+            }
+        }
+        return rulesMap[server.id]!![rule.trim().replace("\\*|\\.".toRegex(), "")]
+    }
```

Comment:
Honestly I'd prefer rules go into a json file, it's easier then servers having to guess the format based on the code. Sorry :/

Reply:
Just make a new config type, probably a HashMap<String, String>

Reply:
Ok so don't parse the strings anymore just have it save in json? Or should I leave that in there to generate a rules.json with info in usage/help?

Reply:
Don't parse the strings, save it as a json